### PR TITLE
FIX constantly displaying time tracker notification

### PR DESCRIPTION
**Pull request by vaniakerer**
https://trello.com/c/RFWCjsBW/66-android-q-the-screen-is-always-turned-on-if-timer-is-running

### CODE REVIEW

File: app/src/main/java/co/techmagic/hr/presentation/time_tracker/HrAppTimeTrackerPresenter.kt [hunk 1]
```diff
@@ -194,9 +194,16 @@ class HrAppTimeTrackerPresenter(
             timeTrackerInteractor.stopTimer().subscribe({ updateReportViewModel(TaskUpdate(it, TaskTimerState.RUNNING)) }, this::showError)
         } else {
             if (canStartReportTracking(userReportViewModel)) {
-                val userReport = userReportViewMadelMapper.retransform(userReportViewModel)
-                timeTrackerInteractor.startTimer(userReport, totalDayMinutesExcludeCurrent(userReport.date.toCalendar(), userReport.minutes))
-                        .subscribe({ updateReportViewModel(TaskUpdate(it.current, TaskTimerState.RUNNING)) }, this::showError)
+                router?.let {
+                    if (!it.isBatteryOptimizationTurnedOn()) {
```
Comment (AlexanderKozubets) [inline app/src/main/java/co/techmagic/hr/presentation/time_tracker/HrAppTimeTrackerPresenter.kt]: As I understood, we should handle this in all places where we're trying to start a timer, right?
Does it make sense to move this check to the service into `startTimer` method and throw an Exception if optimization is turned on? We can handle our custom exception right in the base fragment in `handleError` method.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -9,6 +9,10 @@
     <string name="message_cancel">CANCEL</string>
     <string name="message_delete">DELETE</string>
     <string name="message_unknown_exception">Unknown exception</string>
+
+    <!--Battery Optimization-->
+    <string name="battery_optimization_turn_off_message">We have to turn off battery optimization for correct work</string>
```
Comment (AlexanderKozubets) [inline app/src/main/res/values/strings.xml]: It sounds like a good explanation for a developer, but we need to rephrase it for a user.
I'll get back soon with some suggestions

File: app/src/main/java/co/techmagic/hr/presentation/time_tracker/HrAppTimeTrackerPresenter.kt (hunk 1 above)
Comment (vaniakerer) [reply app/src/main/java/co/techmagic/hr/presentation/time_tracker/HrAppTimeTrackerPresenter.kt]: Agree

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (AlexanderKozubets) [reply app/src/main/res/values/strings.xml]: Please change text to “Do you agree to disable battery optimization for correct work of timer?”

**Review by AlexanderKozubets — APPROVED** [review]

**Pull request merged**